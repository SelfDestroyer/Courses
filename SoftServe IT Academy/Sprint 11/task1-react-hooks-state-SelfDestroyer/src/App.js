import React, { useState } from "react";

import "./App.css";

export default function App() {
  const [text, handleClick] = useState("React Marathon");

  return (
    <div onClick={() => handleClick(text.toLocaleLowerCase())}>{text}</div>
  );
}
