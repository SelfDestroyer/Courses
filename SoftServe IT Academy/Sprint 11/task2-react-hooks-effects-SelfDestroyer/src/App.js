import React, { useEffect, useRef } from "react";

import "./App.css";

export default function App() {
  const textInput = useRef(null);

  useEffect(() => {
    textInput.current.value = localStorage.getItem("appData");
  });

  return (
    <div>
      React Marathon, appData:{" "}
      <input
        size="5"
        onChange={(e) => {
          localStorage.setItem("appData", e.currentTarget.value);
        }}
        ref={textInput}
      ></input>
    </div>
  );
}
