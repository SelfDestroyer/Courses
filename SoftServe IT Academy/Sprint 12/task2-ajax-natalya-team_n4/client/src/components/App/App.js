import Nav from "../Nav/Nav.jsx";
import Persons from "../Persons/Persons.jsx";
import Planets from "../Planets/Planets.jsx";
import Starships from "../Starships/Starships.jsx";
import Films from "../Films/Films.jsx";
import "./App.css";
import { BrowserRouter, Route, Router, Routes } from "react-router-dom";

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Nav />
        <Routes>
          <Route exact path="/planets" element={<Planets />} />
          <Route path="/persons" element={<Persons />} />
          <Route path="/starships" element={<Starships />} />
          <Route path="/films" element={<Films />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
