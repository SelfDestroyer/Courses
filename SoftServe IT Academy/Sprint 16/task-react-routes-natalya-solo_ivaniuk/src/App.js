import React, { useState } from "react";
import { Routes, Route, useNavigate, Link, Navigate } from "react-router-dom";
import "./App.css";
import Subtask1 from "./Components/Subtask1";
import Subtask2 from "./Components/Subtask2";
import Subtask3 from "./Components/Subtask3";
import Subtask4 from "./Components/Subtask4";

export default function App() {
  const navigate = useNavigate();
  const [number, setNumber] = useState(0);

  const test = (e) => setNumber(e.target.value);

  const handleChange = (e) =>
    e.target.checked ? navigate("/subtask4") : navigate("/");

  return (
    <div className="App">
      <h1>React Marathon</h1>
      <h2>The topic 'Routes'</h2>
      <Link to={number % 2 !== 0 ? "/subtask3" : "/"} onClick={test}>
        Show protected information if
      </Link>
      <span>&nbsp;</span>
      <input size="5" onChange={test}></input> is odd
      <div className="mainClass">
        Go to the component programmatically, by checking the box:{" "}
        <input type="checkbox" onChange={handleChange}></input>
      </div>
      <Routes>
        <Route path="/subtask1/:id" element={<Subtask1 />} />
        <Route path="/subtask2" element={<Subtask2 />} />
        <Route path="/subtask3" element={<Subtask3 />} />
        <Route path="/subtask4" element={<Subtask4 />} />
      </Routes>
    </div>
  );
}
