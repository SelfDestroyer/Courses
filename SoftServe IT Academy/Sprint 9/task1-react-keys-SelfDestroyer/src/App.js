import React from "react";
import "./App.css";

export default function App() {
  const data = [
    "Animals",
    "Anime",
    "Anti-Malware",
    "Art Design",
    "Books",
    "Business",
    "Calendar",
    "Cloud Storage",
    "File Sharing",
    "Animals",
    "Continuous Integration",
    "Cryptocurrency",
  ];

  return (
    <div>
      <ul>
        {data.map((el, index) => (
          <li key={String.fromCharCode(index)}>{el}</li>
        ))}
      </ul>
    </div>
  );
}
