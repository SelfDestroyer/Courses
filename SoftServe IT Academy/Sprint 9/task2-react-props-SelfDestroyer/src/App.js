import React from "react";
import "./App.css";
import First from "./First";

export default function App() {
  const data = [
    "Animals",
    "Anime",
    "Anti-Malware",
    "Art Design",
    "Books",
    "Business",
    "Calendar",
    "Cloud Storage",
    "File Sharing",
    "Animals",
    "Continuous Integration",
    "Cryptocurrency",
  ];

  return (
    <div>
      Some data:
      <First list={data.map((el) => el.toLowerCase())} />
    </div>
  );
}
