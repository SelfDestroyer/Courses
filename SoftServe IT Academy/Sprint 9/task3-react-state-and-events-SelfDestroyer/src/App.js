import React from "react";
import "./App.css";

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      appData: "React Marathon",
    };

    this.toLower = this.toLower.bind(this);
  }

  toLower() {
    this.setState({
      appData: this.state.appData.toLowerCase(),
    });
  }

  render() {
    return <div onClick={this.toLower}> {this.state.appData} </div>;
  }
}
