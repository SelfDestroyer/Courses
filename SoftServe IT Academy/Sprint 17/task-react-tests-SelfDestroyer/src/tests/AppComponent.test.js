import React from "react";
import TestRenderer from "react-test-renderer";
import App from "../components/App";

describe("Test component App", () => {
  let componentApp;
  beforeEach(() => {
    componentApp = TestRenderer.create(<App />);
  });

  it("Render corectly", () => {
    expect(componentApp.toJSON()).toMatchSnapshot();
  });

  it("Check type", () => {
    expect(componentApp.toJSON().type).toBe("div");
  });
});
