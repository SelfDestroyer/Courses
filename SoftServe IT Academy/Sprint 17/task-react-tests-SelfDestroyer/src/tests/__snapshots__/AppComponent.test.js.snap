// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component App Render corectly 1`] = `
<div
  className="App"
>
  <h1>
    Hello World!
  </h1>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul>
      <li>
        error
        : 
        request error
      </li>
    </ul>
  </div>
  <div>
    <h3>
      GitHub User Info
    </h3>
    <ul>
      <li>
        error
        : 
        request error
      </li>
    </ul>
  </div>
</div>
`;
